plugin/ensime.vim

```diff
@@ -38,6 +37,10 @@ command! -nargs=* -range EnToggleFullType call ensime#com_en_toggle_fulltype([<f
 command! -nargs=* -range EnOrganizeImports call ensime#com_en_organize_imports([<f-args>], '')
 command! -nargs=* -range EnAddImport call ensime#com_en_add_import([<f-args>], '')
 
+function! EnPackageDecl() abort
+  return ensime#com_en_package_decl()
+endfunction
```

Comment:
Okay this comment is going back into the weeds of off-topic for the pull request, but I'm not sure where to put it yet other than in this conversation context, so I'm just leaving it here and can refer to it later from wherever a better place is…

> `(args, range)` allows us to generalize over all the methods and gives us the ability to customize the external API of a method in an easy way. AFAIK, it is a popular pattern used throughout all the vim plugins and community.

I can sort of understand reasoning for this *if it's for functions that implement commands specifically* (the `com_` ones in our case), but if it starts to proliferate to everything else—as it is in this pull request—I feel it's a bad habit. It's like having a function interface that's mostly meaningless, you can't tell at all how anything should be called unless there is documentation of what the parameters should hold (we have none so far), or else you have to read the code every time to see how they're used within each function. That gets tiresome. We're choosing to make refactoring easier for we the developers, _maybe_, at the expense of poor API for users (at least users who are sophisticated enough to explore the functions that are exposed, not just the documented commands).

I can cite lots of well-written Vim plugins that use meaningful function parameters:

- [Unite](https://github.com/Shougo/unite.vim/blob/master/autoload/unite.vim)
- [lexima](https://github.com/cohama/lexima.vim/blob/master/autoload/lexima.vim)
- [vim-go](https://github.com/fatih/vim-go/blob/master/autoload/go/util.vim)

To name a few that are instructive to read.

> It was present in the plugin even before I started to contribute.

I don't mean to be critical of anyone personally, we have code that works and is useful today, and it got there through your organic improvements. I'm just sharing some thoughts where I've felt there is room for continuing refinement, from working with the code a little bit so far.

It would be better to make some examples via a PR and discuss from there, and I'll try when I have the time, but it's a case where it could end up being a big, out-of-nowhere refactoring PR that wastes a lot of time if we didn't have some discussion beforehand ☺️